static/app/views/discover/table/tableView.tsx

```diff
@@ -190,13 +190,7 @@ function TableView(props: TableViewProps) {
 
       let target;
 
-      if (dataRow.trace === null || dataRow.trace === undefined) {
-        if (dataRow['event.type'] === 'transaction') {
-          throw new Error(
-            'Transaction event should always have a trace associated with it.'
-          );
-        }
-
+      if (dataRow['event.type'] !== 'transaction') {
```

Comment:
Nit: This is more of a stylistic thing but can we invert this condition

```typescript
if (dataRow['event.type'] === 'transaction) {
  // the stuff current in side else block
} else {
  // the stuff currently in the if block
}
```

Makes this easier to reason about at a glance with fewer negations.